src/main/java/com/azure/cosmos/examples/patch/sync/SamplePatchQuickstart.java

```diff
@@ -0,0 +1,490 @@
+
+public class SamplePatchQuickstart {
+
+    private CosmosClient client;
+
+    private final String databaseName = "AzureSampleFamilyDB";
+    private final String containerName = "FamilyContainer";
+
+    private CosmosDatabase database;
+    private CosmosContainer container;
+
+    protected static Logger logger = LoggerFactory.getLogger(SamplePatchQuickstart.class);
+
+    public void close() {
+        client.close();
+    }
+
+    /**
+     * Run a Hello CosmosDB console application.
+     * <p>
+     * This is a simple sample application intended to demonstrate Create, Read,
+     * Update, Delete (CRUD) operations with Azure Cosmos DB Java SDK, as applied to
+     * databases, containers and items. This sample will 1. Create synchronous
+     * client, database and container instances 2. Create several items 3. Upsert
+     * one of the items 4. Perform a query over the items 5. Delete an item 6.
+     * Delete the Cosmos DB database and container resources and close the client. *
+     */
+    // <Main>
+    public static void main(String[] args) {
+        SamplePatchQuickstart p = new SamplePatchQuickstart();
+
+        try {
+            logger.info("Starting SYNC main");
+            p.getStartedDemo();
+            logger.info("Demo complete, please hold while resources are released");
+        } catch (Exception e) {
+            e.printStackTrace();
+            logger.error(String.format("Cosmos getStarted failed with %s", e));
+        } finally {
+            logger.info("Closing the client");
+            p.shutdown();
+        }
+    }
+
+    // </Main>
+
+    private void getStartedDemo() throws Exception {
+
+        logger.info("Using Azure Cosmos DB endpoint: " + AccountSettings.HOST);
+
+        ArrayList<String> preferredRegions = new ArrayList<String>();
+        preferredRegions.add("West US");
+
+        // Create sync client
+        // <CreateSyncClient>
+        client = new CosmosClientBuilder().endpoint(AccountSettings.HOST).key(AccountSettings.MASTER_KEY)
+                .preferredRegions(preferredRegions).consistencyLevel(ConsistencyLevel.EVENTUAL)
+                .contentResponseOnWriteEnabled(true).buildClient();
+
+        // </CreateSyncClient>
+
+        createDatabaseIfNotExists();
+        createContainerIfNotExists();
+
+        // Setup family items to create
+        ArrayList<Family> familiesToCreate = new ArrayList<>();
+        familiesToCreate.add(Families.getAndersenFamilyItem());
+        familiesToCreate.add(Families.getWakefieldFamilyItem());
+        familiesToCreate.add(Families.getJohnsonFamilyItem());
+        familiesToCreate.add(Families.getSmithFamilyItem());
+
+        // Creates several items in the container
+        // Also applies an upsert operation to one of the items (create if not present,
+        // otherwise replace)
+        createFamilies(familiesToCreate);
+
+        patchAddSingle(familiesToCreate.get(0).getId(), familiesToCreate.get(0).getLastName());
```

Comment:
why don't we pass the Family object and get the id and Last name inside the function.
This arguments outside also looks good to me but still just a discussion.

Reply:
That's a good point. I kept it this way so that the function signature remains self-explanatory:

`private void patchAddSingle(String id, String partitionKey){...}`

For e.g. here we know that `id` and `partitionKey` is being used for patch operation as opposed to a `Family` object which *might* seem a little obscure.

Let me know if you still feel I need to change all the functions to use a `Family` object